src/megaapi_impl.cpp

```diff
@@ -22863,6 +23382,29 @@ void MegaSyncPrivate::setLocalFolder(const char *path)
     localFolder =  MegaApi::strdup(path);
 }
 
+const char *MegaSyncPrivate::getMegaFolder() const
+{
+    return megaFolder;
+}
+
+void MegaSyncPrivate::setMegaFolder(const char *path)
+{
+    if (megaFolder)
+    {
+        delete [] megaFolder;
+    }
+    megaFolder = MegaApi::strdup(path);
+}
+
+void MegaSyncPrivate::setMegaFolderYielding(char *path)
```

Comment:
I was precisely trying to avoid a second allocation, hence this yielding method.
The header indicates that:
```     void setMegaFolderYielding(char *path); //MEGAsync acquires the ownership of path
```
let me know if that's not enough and will change it afer merging https://github.com/meganz/sdk/pull/2091